### PR TITLE
SUBS 519 dispatcher performance

### PR DESCRIPTION
**Pull request by davidrichardson**

### CODE REVIEW

**Comment by davidrichardson** [comment]
I've tried to refactor the dispatcher to improve performance and make it work systematically, for any archive / data type. Please give this careful review.

File: subs-processing-model/src/main/java/uk/ac/ebi/subs/processing/SubmissionEnvelope.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package uk.ac.ebi.subs.processing;
 
 
+import jdk.nashorn.internal.runtime.StoredScript;
```
Comment (fpenim) [inline subs-processing-model/src/main/java/uk/ac/ebi/subs/processing/SubmissionEnvelope.java]: This doesn't look like it's being used and my IDE can't even find it.

File: subs-repository/src/main/java/uk/ac/ebi/subs/repository/repos/status/ProcessingStatusRepositoryImpl.java [hunk 2]
```diff
@@ -77,6 +74,42 @@ public ProcessingStatusRepositoryImpl(MongoTemplate mongoTemplate) {
...
+        return idsByType;
+
+/*
+* db.getCollection('processingStatus').aggregate(
+{"$match": {"submissionId": "e397d8e9-3b8b-4093-b82c-e23ae5333179"}},
+{"$match": {"status": {"$in": ["Draft","Submitted"]}}},
+{"$group": {"_id": "$submittableType", "submittableIds": {"$addToSet": "$submittableId"}}},
+{"$project": {"submittableType": "$_id", "submittableIds": "$submittableIds"}}
+)
+* */
```
Comment (fpenim) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/repos/status/ProcessingStatusRepositoryImpl.java]: What is the purpose of this comment? Should it be in a JavaDoc?

File: subs-repository/src/main/java/uk/ac/ebi/subs/repository/services/SubmittableHelperService.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        setTeamFromSubmission(submittable);
+    }
+
+    private void createValidationResult(StoredSubmittable submittable) {
+        ValidationResult validationResult = new ValidationResult();
+        validationResult.setEntityUuid(submittable.getId());
+        validationResult.setUuid(UUID.randomUUID().toString());
+
+        if (submittable.getSubmission() != null)
```
Comment (fpenim) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/services/SubmittableHelperService.java]: Can we have a submittable not being in a Submission?

Comment (davidrichardson) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/services/SubmittableHelperService.java]: Not really, but spring data rest validates after the 'before' events have been handled, so we will see submittables at this stage without a submission. I will document this.

File: subs-processing-model/src/main/java/uk/ac/ebi/subs/processing/SubmissionEnvelope.java (hunk 1 above)
Comment (davidrichardson) [inline subs-processing-model/src/main/java/uk/ac/ebi/subs/processing/SubmissionEnvelope.java]: Removed.

File: subs-repository/src/main/java/uk/ac/ebi/subs/repository/repos/status/ProcessingStatusRepositoryImpl.java (hunk 2 above)
Comment (davidrichardson) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/repos/status/ProcessingStatusRepositoryImpl.java]: I will remove this, it's my working notes from making the aggregate query

**Pull request merged**